content/en/docs/concepts/overview/kubernetes-api.md

```diff
@@ -22,63 +22,131 @@ external components communicate with one another.
 The Kubernetes API lets you query and manipulate the state of API objects in Kubernetes
 (for example: Pods, Namespaces, ConfigMaps, and Events).
 
-Most operations can be performed through the
-[kubectl](/docs/reference/kubectl/) command-line interface or other
-command-line tools, such as
-[kubeadm](/docs/reference/setup-tools/kubeadm/), which in turn use the
-API. However, you can also access the API directly using REST calls.
+Most operations can be performed through the [kubectl](/docs/reference/kubectl/)
+command-line interface or other command-line tools, such as
+[kubeadm](/docs/reference/setup-tools/kubeadm/), which in turn use the API.
+However, you can also access the API directly using REST calls. Kubernetes
+provides a set of [client
+libraries](/docs/reference/using-api/client-libraries/) for those looking to
+write applications using the Kubernetes API. 
+
+Kubernetes offers two "machine readable" APIs for listing all other Kubernetes
+APIs. Both are valuable tools for developers who are building client libraries,
+tooling, IDE features, documentation, and other tools interacting with
+Kubernetes.
+
+Discovery is an API that provides information about available API names,
```

Comment:
```suggestion
[Discovery](#discovery-api) is an API that provides information about available API names,
```

Reply:
I would also mention that this is a kubernetes thing that doesn't exist somewhere else. That term is used by many different systems and can cause confusion.